internal/xpkg/cache.go

```diff
@@ -17,95 +17,103 @@ limitations under the License.
 package xpkg
 
 import (
+	"compress/gzip"
 	"io"
 	"os"
 	"sync"
 
-	"github.com/google/go-containerregistry/pkg/name"
-	v1 "github.com/google/go-containerregistry/pkg/v1"
-	"github.com/google/go-containerregistry/pkg/v1/tarball"
 	"github.com/spf13/afero"
 
 	"github.com/crossplane/crossplane-runtime/pkg/errors"
 )
 
 const (
-	errGetNopCache = "cannot get an image from a NopCache"
+	errGetNopCache = "cannot get content from a NopCache"
 )
 
-// A Cache caches OCI images.
+const cacheContentExt = ".gz"
+
+// A Cache caches package content.
 type Cache interface {
-	Get(tag string, id string) (v1.Image, error)
-	Store(tag string, id string, img v1.Image) error
+	Has(id string) bool
+	Get(id string) (io.ReadCloser, error)
+	Store(id string, content io.ReadCloser) error
 	Delete(id string) error
 }
 
-// ImageCache stores and retrieves OCI images in a filesystem-backed cache in a
-// thread-safe manner.
-type ImageCache struct {
+// PackageCache stores and retrieves package content in a filesystem-backed
+// cache in a thread-safe manner.
```

Comment:
I imagine changing from a cache of packages to a cache of YAML streams has some implications:

1. We're slightly more committed to "the content of a Crossplane package" as far as the things the package manager needs to operate on a package being a single YAML stream. Probably not a problem?
2. Folks updating to this implementation will presumably have their package caches invalidated. Is It worth adding logic to cleanup the old image based cache? IMO probably not.

Reply:
1. I don't think moving to caching just the YAML stream _necessarily_ prohibits us from extending the package format in the future -- all packages would just need to be pulled again, which shouldn't be a huge issue.
2. I thought about this, but given that there is not a possibility for conflict with content previously written by the cache, I agree that the added complexity of trying to clean up is not worth it. One area I do want to note in the release notes is that if you had manually populated the cache, you'll need to repopulate it again with a different process. I don't think this is a common use case today though.